src/engine/resource_manager/loader/curve.rs

```diff
@@ -10,15 +11,14 @@ use crate::{
 pub struct CurveLoader;
 
 impl ResourceLoader<CurveResource, CurveImportOptions> for CurveLoader {
-    type Output = BoxedLoaderFuture;
-
     fn load(
-        &mut self,
+        &self,
         curve: CurveResource,
         _default_import_options: CurveImportOptions,
+        _resource_manager: ResourceManager,
```

Comment:
Do not pass resource manager to `load`, this was removed in one of latest commits. Loaders should acquire their own copy of ResourceManager explicitly, as it was already done for ModelLoader.

Reply:
> Do not pass resource manager to `load`, this was removed in one of latest commits. Loaders should acquire their own copy of ResourceManager explicitly, as it was already done for ModelLoader.

That would be an issue during ResourceManager creation: In the current API resource manager is created later as the loaders. In the non-configurable current solution in the new The resource manager is crated than the loaders and finally they are "glued" together (https://github.com/FyroxEngine/Fyrox/blob/c24bfa5679161625c7edad07d7df93cb75979522/src/engine/resource_manager/mod.rs#L97). 
So either the whole state setup have to be moved outside of ResourceManager::new  or ResourceManager may require functions to alter the loaders once it's been created. The latter may align better to provide a ResourceManager instead of a builder for the Engine.

Reply:
Yes, a set of functions to replace loaders is a good alternative.